Move emphasis from "Any" to "object"

DanKaplanSES opened the pull request:
Before:

> **Any** object, including a `Boolean` object whose value is `false`, evaluates to `true` when passed to a conditional statement.

After:

> Any **object**, including a `Boolean` object whose value is `false`, evaluates to `true` when passed to a conditional statement.

DanKaplanSES committed:
Move emphasis from "Any" to "object"

- My intent is to establish early that that sentence that follows pertains to Boolean objects, not primitives.
- Whether this is an improvement is highly subjective, but given the section is titled "Boolean primitives and Boolean objects," it seemed to me that the word "object" should stand out more than the word "Any."

dipikabh commented on files/en-us/web/javascript/reference/global_objects/boolean/index.md:
```diff
@@ -15,7 +15,7 @@ The **`Boolean`** object represents a truth value: `true` or `false`.
 Do not confuse the {{Glossary("Primitive", "primitive")}} `Boolean` values `true` and `false` with the `true` and `false` values of the `Boolean` object.
 
-**Any** object, including a `Boolean` object whose value is `false`, evaluates to `true` when passed to a conditional statement. For example, the condition in the following {{jsxref("Statements/if...else", "if")}} statement evaluates to `true`:
+Any **object**, including a `Boolean` object whose value is `false`, evaluates to `true` when passed to a conditional statement. For example, the condition in the following {{jsxref("Statements/if...else", "if")}} statement evaluates to `true`:
```
The original emphasis is correct because it conveys that this applies to all objects. Emphasizing "object" shifts this focus.
The sentence itself is pretty straightforward and adding the emphasis is actually distracting. Perhaps it will be better to remove the emphasis completely.
```suggestion
Any object, including a `Boolean` object whose value is `false`, evaluates to `true` when passed to a conditional statement. For example, the condition in the following {{jsxref("Statements/if...else", "if")}} statement evaluates to `true`:
```

DanKaplanSES replied:
Hm, I didn't realize when I created this PR, but I don't think I understand what the sentence in question is intending to say. 

1. I assumed it meant, "Any object, as opposed to a primitive boolean" or "Any objects, as opposed to any primitive." Am I wrong?
1. What objects besides `Boolean` can have a value of `false`; the sentence says, "Any object ... whose value is `false`". 

---

Here's the original version for convenience:

> ### Boolean primitives and Boolean objects
> 
> Do not confuse the {{Glossary("Primitive", "primitive")}} `Boolean` values `true` and `false` with the `true` and `false` values of the `Boolean` object.
> 
> **Any** object, including a `Boolean` object whose value is `false`, evaluates to `true` when passed to a conditional statement. For example, the condition in the following {{jsxref("Statements/if...else", "if")}} statement evaluates to `true`:
> 
> ```js
> const x = new Boolean(false);
> if (x) {
>   // this code is executed
> }
> ```
>

dipikabh reviewed (changes requested):
Thanks for the PR, @DanKaplanSES.

DanKaplanSES committed:
Rewrote "Boolean primitives and Boolean objects"

- On my first attempt I didn't truly realize what the line I changed was articulating until I received feedback. Here is my second attempt at this PR.
- This change may clash with the tone of the rest of the article because it is written in a more casual voice, but if that's an issue, I'd be happy to change that if there's potential behind the edit.
- If this is not an improvement, I'm okay with closing this PR without a merge.

DanKaplanSES commented:
I rewrote the first half of this subsection but it may not be an improvement over the original. I'm okay closing this without comment :)

DanKaplanSES committed:
Update files/en-us/web/javascript/reference/global_objects/boolean/index.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

Josh-Cena committed:
Better wording

Josh-Cena reviewed (approved):
I think the original text is too roundabout and doesn't get to the point fast enough. Rewrote another version. Let me know what you think.

DanKaplanSES commented:
> I think the original text is too roundabout and doesn't get to the point fast enough.

I think I've been making false assumptions: what _is_ the point it's trying to get to? I thought it was the difference between Boolean primitives and Boolean objects.

Josh-Cena commented:
The point is to not use objects. My revision makes it the first sentence.

dipikabh commented on files/en-us/web/javascript/reference/global_objects/boolean/index.md:
````diff
@@ -13,26 +13,6 @@ The **`Boolean`** object represents a truth value: `true` or `false`.
-}
-```
-
 Do not use the `Boolean()` constructor with `new` to convert a non-boolean value to a boolean value — use `Boolean` as a function or a [double NOT](/en-US/docs/Web/JavaScript/Reference/Operators/Logical_NOT#double_not_!!) instead:
````
How about we first state how it should be done and then what should be avoided?
```suggestion
For converting non-boolean values to boolean, use `Boolean` as a function or use the [double NOT](/en-US/docs/Web/JavaScript/Reference/Operators/Logical_NOT#double_not_!!) operator. Do not use the `Boolean()` constructor with `new`.
```

dipikabh commented on files/en-us/web/javascript/reference/global_objects/boolean/index.md:
````diff
@@ -13,26 +13,6 @@ The **`Boolean`** object represents a truth value: `true` or `false`.
-
 Do not use the `Boolean()` constructor with `new` to convert a non-boolean value to a boolean value — use `Boolean` as a function or a [double NOT](/en-US/docs/Web/JavaScript/Reference/Operators/Logical_NOT#double_not_!!) instead:
 
 ```js
````
can we rewrite this example using good and bad info strings:

````
```js example-good
const good = Boolean(expression); // use this
const good2 = !!expression; // or use this
```
```js example-bad
const bad = new Boolean(expression); // don't use this
```
````

dipikabh commented on files/en-us/web/javascript/reference/global_objects/boolean/index.md:
````diff
@@ -41,13 +21,21 @@ const good2 = !!expression; // or this
 ```
 
-If you specify any object, including a `Boolean` object whose value is `false`, as the initial value of a `Boolean` object, the new `Boolean` object has a value of `true`.
+This is because _all_ objects, including a `Boolean` object whose wrapped value is `false`, are {{glossary("truthy")}} and evaluate to `true` in places such as conditional statements. (See also the [boolean coercion](#boolean_coercion) section below.)
````
- Will all readers understand "wrapped value"?
- Would it help to add another sentence here:
```suggestion
This is because _all_ objects in JavaScript, including a `Boolean` object with a `false` value, are considered {{glossary("truthy")}} and evaluate to `true` in conditional contexts. This means the object's underlying Boolean value (`true` or `false`) does not affect its truthy evaluation in conditions like `if` statements. (See also the [boolean coercion](#boolean_coercion) section below.)
```

Josh-Cena replied:
> Will all readers understand "wrapped value"?

It's still technically much more accurate than "with a `false` value", because an object cannot have a primitive "as its value". It can only "wrap" that value. I don't think "wrapped value" and "underlying value" have any meaningful difference in comprehension.

dipikabh reviewed (changes requested):
Thanks for the rewrite, @Josh-Cena. I think this version does get to the point quickly. I've added a few suggestions.

Josh-Cena committed:
Update files/en-us/web/javascript/reference/global_objects/boolean/index.md

Co-authored-by: Dipika Bhattacharya <dipika@foss-community.org>

Josh-Cena committed:
Update index.md

dipikabh reviewed (approved):
Thanks, @Josh-Cena 👍

DanKaplanSES commented:
@Josh-Cena 
> The point is to not use objects. My revision makes it the first sentence.

Ohhh! I was confused because the section is titled, "Boolean primitives and Boolean objects." This whole time I thought the point was to describe the differences, and "don't use Boolean objects" would be the/a take away. 

And with that last bit of understanding, I think I've finally caught up with everyone else in this thread :) Thanks for your patience!
